File: 3372-longest-strictly-increasing-or-strictly-decreasing-subarray/solution.cpp
class Solution {
public:
    int longestMonotonicSubarray(vector<int>& nums) {
        int inc = 1, dec = 1, res = 1;

        // run a single pass iteration
        for (int i = 0; i < nums.size() - 1; i++) {
            if (nums[i+1] > nums[i]) {
                // check for strictly increasing sequence
                inc++;
                dec = 1; // reset dec
            } else if (nums[i+1] < nums[i]) {
                // check for strictly decreasing sequence
                dec++;
                inc = 1; // reset inc
            } else {
                // reset counters if none SI or SD
                inc = dec = 1;
            }

            // finally get the max out of the 3 counters
            res = max({res, inc, dec});
        }

        return res;
    }
};

File: 3372-longest-strictly-increasing-or-strictly-decreasing-subarray/solution.py
class Solution:
    def longestMonotonicSubarray(self, nums: List[int]) -> int:
        inc = dec = res = 1

        for i in range(len(nums) - 1):
            if nums[i+1] > nums[i]:
                # next number is larger, SI sequence
                inc += 1
                dec = 1 # reset
            elif nums[i+1] < nums[i]:
                dec += 1
                inc = 1
            else:
                # reset both counters
                inc = dec = 1
            
            # lastly, update result for each iteration
            res = max(res, inc, dec)

        return res
